File: Blind 75 LeetCode/Chahat Rajput/Coin Change/coin_change.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int>dp(amount+1,amount+1);
        dp[0]=0;
        for(int i=1;i<=amount;i++){
            for(int j=0;j<coins.size();j++){
                if(i>=coins[j])
                    dp[i]=min(dp[i-coins[j]]+1,dp[i]);
            }
        }
       return dp[amount]>amount ?-1:dp[amount];
    }
};

File: Blind 75 LeetCode/Subhangi dhasmana/Coin change/Coin change.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        dp = [0] + ([float('inf')] * amount)
        for i in range(1, amount + 1):
            for coin in coins:
                if coin <= i:
                    dp[i] = min(dp[i], dp[i - coin] + 1)

        if dp[-1] == float('inf'):
            return -1
        return dp[-1]